src/modules/fw_pos_control_l1/fw_pos_control_l1_main.cpp

```diff
@@ -2283,6 +2289,23 @@ FixedwingPositionControl::task_main()
 			/* load local copies */
 			orb_copy(ORB_ID(vehicle_global_position), _global_pos_sub, &_global_pos);
 
+			// handle estimator reset events. we only adjust setpoins for manual modes
+			if (_control_mode.flag_control_manual_enabled) {
+				if (_control_mode.flag_control_altitude_enabled && _global_pos.alt_reset_counter != _alt_reset_counter) {
+					_hold_alt += _global_pos.delta_alt;
```

Comment:
@julianoes That counter should not be reset! The ekf increased the counter after every reset and the position controller should track it.